app/src/main/java/com/github/sdpteam15/polyevents/database/FirestoreDatabaseProvider.kt

```diff
@@ -0,0 +1,207 @@
+package com.github.sdpteam15.polyevents.database
+
+import com.github.sdpteam15.polyevents.database.observe.Observable
+import com.github.sdpteam15.polyevents.event.Event
+import com.github.sdpteam15.polyevents.user.ProfileInterface
+import com.github.sdpteam15.polyevents.user.UserInterface
+import com.google.android.gms.tasks.OnFailureListener
+import com.google.android.gms.tasks.OnSuccessListener
+import com.google.android.gms.tasks.Task
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.firestore.DocumentSnapshot
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.firestore.QuerySnapshot
+import com.google.firebase.firestore.ktx.firestore
+import com.google.firebase.ktx.Firebase
+
+
+object FirestoreDatabaseProvider : DatabaseInterface {
+    var firestore: FirebaseFirestore? = null
+        get() = field ?: Firebase.firestore
+
+    const val USER_DOCUMENT = "users"
+    const val USER_DOCUMENT_ID = "uid"
+    const val EVENT_DOCUMENT = "events"
+    const val EVENT_DOCUMENT_ID = "eventId"
+    const val ITEM_DOCUMENT = "items"
+    const val ITEM_DOCUMENT_ID = "itemId"
+    const val AREA_DOCUMENT = "areas"
+    const val AREA_DOCUMENT_ID = "areaId"
+
+
+    val firstConnectionMap = HashMap<String, String>()
```

Comment:
missing comment

Reply:
True, I add them so that it's more clear what this map represent